src/utilities.ts

```diff
@@ -50,6 +50,9 @@ export const yearRange = 6;
 
 export function getCollegeFullName(acronym: string | undefined): string {
   // Return empty string if college is not in requirementJSON
+  if (acronym === 'AS') {
```

Comment:
This question is totally fair, and it's due to an inconsistency with the frontend and "backend" (how we represent requirements data).

A&S before and after 2020 have different requirements. They are represented as `AS1` and `AS2` in our requirements data. That is, they are considered separate colleges (because we assume one college = one set of requirements). On the frontend, we want the user to see it as a merged college, i.e. `AS`. So, Zak decided to implement this as a frontend-only change by writing conversions between these two representations. An upside of this is that the backend format can stay the same. A downside of this is that A&S needs to be special-cased in few different places.

An alternative implementation would have been something along the lines of adding variants to the requirements data format. So, the backend would look something like `AS:Version1` and `AS:Version2`, where `AS` refers to a single college and `Version1` and `Version2` refer to different sets of requirements. An upside would be that it would be more extensible -- that is, other colleges can have multiple versions as well (which is great if requirements are changed relatively frequently). A downside would be that every other college would have to support this added complexity (for example, `EN` would need to turn into `EN:Version1` or `EN:Default`).

I personally think a single special case like this is fine. But if there are more special cases to be added in the future, or if we somehow manage to reach the backlog, I think there should be a refactoring effort.

Reply:
Thanks for explaining that Ben! If we ever have a chance to refactor this to use the alternate implementation you mentioned, that could allow us to handle more specific cases of requirements changing over time (such as the [Probability and Stats requirement for ECE](https://github.com/cornell-dti/course-plan/pull/612#discussion_r767367196)). It's definitely down the line, but it could be useful for our users and help to reduce some confusion.